src/Microsoft.Azure.WebJobs.Host/Diagnostics/ExceptionFormatter.cs

```diff
@@ -26,92 +26,108 @@ public static class ExceptionFormatter
-                : className + " : " + message;
+                : className + " : " + message);
             var innerException = exception.InnerException;
             if (innerException != null)
             {
                 if (includeMessageOnly)
                 {
                     do
                     {
-                        stackText += " ---> " + innerException.Message;
+                        sb.Append(" ---> ");
+                        sb.Append(innerException.Message);
                         innerException = innerException.InnerException;
                     }
                     while (innerException != null);
                 }
                 else
                 {
-                    stackText += $" ---> {GetFormattedException(innerException)} {Environment.NewLine}   End of inner exception";
+                    sb.Append(" ---> ");
+                    GetFormattedException(sb, innerException);
+                    sb.AppendLine();
+                    sb.Append("   End of inner exception");
                 }
             }
 
-            string stackTrace = GetAsyncStackTrace(exception);
-            if (!string.IsNullOrEmpty(stackTrace))
-            {
-                stackText += Environment.NewLine + stackTrace;
-            }
-
-            return stackText;
+            AddAsyncStackTrace(sb, exception);
         }
 
-        private static string GetAsyncStackTrace(Exception exception)
+        private static void AddAsyncStackTrace(StringBuilder sb, Exception exception)
         {
             var stackTrace = new StackTrace(exception, fNeedFileInfo: true);
             var stackFrames = stackTrace.GetFrames();
             if (stackFrames == null)
             {
-                return string.Empty;
+                return;
             }
 
-            var stackFrameLines = from frame in stackFrames
-                                  let method = frame.GetMethod()
-                                  let declaringType = method?.DeclaringType
-                                  where ShouldShowFrame(declaringType)
-                                  select FormatFrame(method, declaringType, frame);
-
-            return string.Join(Environment.NewLine, stackFrameLines);
+            foreach (var frame in stackFrames)
+            {
+                FormatFrame(sb, frame);
+            }
         }
 
         private static bool ShouldShowFrame(Type declaringType) =>
             !(declaringType != null && typeof(INotifyCompletion).IsAssignableFrom(declaringType));
 
-        private static string FormatFrame(MethodBase method, Type declaringType, StackFrame frame)
+        private static void FormatFrame(StringBuilder stringBuilder, StackFrame frame)
         {
-            var stringBuilder = new StringBuilder();
+            var method = frame.GetMethod();
+            var declaringType = method?.DeclaringType;
+            
+            if (ShouldShowFrame(declaringType) == false)
```

Comment:
This is a really nitpicky comment :-) -- but can you make this `if (!ShouldShowFrame(declaringType))`?

Reply:
Fixed in https://github.com/Azure/azure-webjobs-sdk/pull/2510/commits/4b6d6d71dbaf5894c1f17ac6598173ab5ef0e5fb